### PR TITLE
Directly suggest `forM_ x y` given `void (forM x y)`

### PR DESCRIPTION
**Pull request by zliu41**
Currently the suggestion for `void (forM x y)` is `void (forM_ x y)`. This is kind of weird if not confusing - if we are keeping the `void`, then why bother changing `forM` to `forM_`. This PR makes HLint directly suggest `(forM_ x y)`.

### CODE REVIEW

File: src/Hint/Monad.hs [hunk 1]
```diff
@@ -45,7 +51,9 @@ main = f $ do g a $ sleep 10 --
 main = do f a $ sleep 10 -- @Ignore
 main = do foo x; return 3; bar z --
 main = void $ forM_ f xs -- forM_ f xs
-main = void $ forM f xs -- void $ forM_ f xs
+main = void (forM_ f xs) -- (forM_ f xs)
```
Comment (ndmitchell) [inline src/Hint/Monad.hs:54]: These brackets seem a bit odd?

**Review by ndmitchell — COMMENTED** [review]
Are the brackets a necessity, or could we drop them?

File: src/Hint/Monad.hs (hunk 1 above)
Comment (zliu41) [reply src/Hint/Monad.hs:54]: Probably, but this is orthogonal and unrelated to this change. `void (forM_ f xs)` -> `(forM_ f xs)` is the same as the previous behavior. The new behavior is `void (forM f xs)` -> `(forM_ f xs)`.

Comment (zliu41) [reply src/Hint/Monad.hs:54]: @ndmitchell I hoped this addresses your question - let me know if you have any other concerns!

Comment (ndmitchell) [reply src/Hint/Monad.hs:54]: Ah, didn't realise that was what it did before. Have fixed that separately in accc63710b17bf9b533288b3c0ba06ecd5c3be06

**Pull request merged**